From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 20 Aug 2023 03:33:52 +0200
Subject: ci: Include .desktop files in POTFILES check

.desktop files always include translatable strings, so make sure
they are added to either POTFILES.in or POTFILES.skip.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2904>
(cherry picked from commit 168b306ffe2cd9a4cf6779451c81c8b426c6be28)
---
 .gitlab-ci/check-potfiles.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitlab-ci/check-potfiles.sh b/.gitlab-ci/check-potfiles.sh
index 82f101c..0969da1 100755
--- a/.gitlab-ci/check-potfiles.sh
+++ b/.gitlab-ci/check-potfiles.sh
@@ -2,6 +2,7 @@
 
 srcdirs="src subprojects/extensions-tool"
 uidirs="js subprojects/extensions-app"
+desktopdirs="data subprojects/extensions-app/ subprojects/extensions-tool"
 
 # find source files that contain gettext keywords
 files=$(grep -lR --include='*.c' '\(gettext\|[^I_)]_\)(' $srcdirs)
@@ -9,6 +10,9 @@ files=$(grep -lR --include='*.c' '\(gettext\|[^I_)]_\)(' $srcdirs)
 # find ui files that contain translatable string
 files="$files "$(grep -lRi --include='*.ui' 'translatable="[ty1]' $uidirs)
 
+# find .desktop files
+files="$files "$(find $desktopdirs -name '*.desktop*')
+
 # filter out excluded files
 if [ -f po/POTFILES.skip ]; then
   files=$(for f in $files; do ! grep -q ^$f po/POTFILES.skip && echo $f; done)
